From e1524fa5630296fae7889fb61ed85715fbc67f8d Mon Sep 17 00:00:00 2001
From: Alona SolntsevaX <asolntsevax@asolntsevax-vm-u20.habana-labs.com>
Date: Thu, 03 Feb 2022 14:57:28 +0200
Subject: [PATCH] [SW-67906] erom: define new hl_fw_comp for erom

Define fw_comp on fwupd_get_proto_status call in Makefile
It will be:
 - HL_FW_COMP_EROM for erom_recovery image
 - HL_FW_COMP_PREBOOT for preboot image
 - HL_FW_COMP_NONE for other images (shouldn't happen)

Change-Id: I9cf335b7858da42ec67619c5a331a9b74b741f1e
Signed-off-by: Alona SolntsevaX <asolntsevax@asolntsevax-vm-u20.habana-labs.com>
---

diff --git a/Makefile b/Makefile
index b783942..514c82c 100644
--- a/Makefile
+++ b/Makefile
@@ -21,6 +21,7 @@
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c -fno-builtin
 CFLAGS += -I ./include/ -I ./ -I ./lib/ -I ./lib/common/ -I ./$(HL_DEV_ARCH)/ -I ./$(HL_DEV_ARCH_COMMON)/
+CFLAGS += -D DEF_HL_FW_COMP_PREBOOT
 
 ifeq ($(HL_TARGET),pldm)
 	CFLAGS += -DTARGET_IS_PLDM
diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 14d7816..9252afe 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -15,6 +15,7 @@
 CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(PREBOOT_ROOT_DIR)/lib/
 CFLAGS += -I $(HL_DEV_ARCH)/ -I $(HL_DEV_ARCH)/../common/ -I $(SPECS_EMBEDDED_DIR)/$(BASE_DEV_NAME)/
 CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(SPECS_DEV)/$(SPECS_DEV)_reg_map.h
+CFLAGS += -D DEF_HL_FW_COMP_EROM
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
 SFLAGS := $(CFLAGS) -D__ASSEMBLY__
diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index 64043ff..e5bf7af 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -120,7 +120,13 @@
 {
 	status->version = HL_UPD_PROTO_VERSION;
 	status->compat_mask = cpu_boot_dev_sts_get_all();
+#if defined(DEF_HL_FW_COMP_EROM)
+	status->fw_comp = HL_FW_COMP_EROM;
+#elif defined(DEF_HL_FW_COMP_PREBOOT)
 	status->fw_comp = HL_FW_COMP_PREBOOT;
+#else
+	status->fw_comp = HL_FW_COMP_NONE;
+#endif
 	status->err = err;
 	status->sub_err = serr;
 }
